src/hooks/useRequestParams.ts

```diff
@@ -1,93 +1,108 @@
-import { useMemo } from "react";
+import assert from "assert";
 import { useSearchParams } from "react-router-dom";
+import * as ss from "superstruct";
```

Comment:
im replacing the previous parsing logic which much safer and simpler version using superstruct